lib/rules/validate-quote-marks.js

```diff
@@ -106,10 +106,16 @@ module.exports.prototype = {
                     return;
                 }
 
+                token.value = token.value.replace(
+                    new RegExp('^' + mark + '(.*)' + mark + '$'),
```

Comment:
That seems overly complicated and slow for what it's doing (which is basically `token.value = opposite[mark] + token.value.slice(1, -1) + opposite[mark]`), and what it's doing seems inaccurate:
* as noted, it will fail on multi-line strings because `.` doesn't match newline (jQuery uses `[\w\W]` for a true "any character" match)
* it will also fail on strings containing unescaped quote characters, e.g. converting raw `"'"` into `'''`

I _**think**_ what you want to do is escape all unescaped quote characters _in_ the string, then change the quote characters _of_ the string... something like:
```js
var unescapedOpposite = {
	'"': /'/g,
	"'": /"/g
};
…
token.value = opposite[mark] +
	str.slice(1, -1).split(/((?:\\[\w\W])+)/).map(function(part, i) {
		return i % 2 === 0 ?
			part.replace(unescapedOpposite[mark], "\\$&") :
			part;
	}).join("") +
	opposite[mark];
```

It's also possible to do with a probably-faster but arguably more complex single-pass `replace`:
```js
var unescapedOpposite = {
	'"': /((?:\\[\w\W]|[^\\])*?)(')/g,
	"'": /((?:\\[\w\W]|[^\\])*?)(")/g
};
…
var start = 0;
token.value = opposite[mark] +
	token.value.slice(1, -1).replace(unescapedOpposite[mark], function(match, prefix, mark, offset) {
		var escaped = offset === start ?
			prefix + "\\" + mark :
			match;
		start = offset + match.length;
		return escaped;
	}) +
	opposite[mark];
```